diff --git a/content/browser/permissions/permission_service_impl.cc b/content/browser/permissions/permission_service_impl.cc
index 064107bb01905b192c9410dd7e9937a74d64afc1..1a568cfe3e2a14384edc21846ed36d16fc2359a2 100644
--- a/content/browser/permissions/permission_service_impl.cc
+++ b/content/browser/permissions/permission_service_impl.cc
@@ -88,7 +88,10 @@ bool PermissionDescriptorToPermissionType(
     case PermissionName::IDLE_DETECTION:
       *permission_type = PermissionType::IDLE_DETECTION;
       return true;
-  }
+    case PermissionName::AUTOPLAY:
+      *permission_type  = PermissionType::AUTOPLAY;
+      return true;
+}
 
   NOTREACHED();
   return false;
