diff --git a/content/browser/permissions/permission_service_impl.cc b/content/browser/permissions/permission_service_impl.cc
index 064107bb01905b192c9410dd7e9937a74d64afc1..50ad5f27301051d16e6bace2a0c70a0da0c02669 100644
--- a/content/browser/permissions/permission_service_impl.cc
+++ b/content/browser/permissions/permission_service_impl.cc
@@ -88,6 +88,9 @@ bool PermissionDescriptorToPermissionType(
     case PermissionName::IDLE_DETECTION:
       *permission_type = PermissionType::IDLE_DETECTION;
       return true;
+    case PermissionName::AUTOPLAY:
+      *permission_type  = PermissionType::AUTOPLAY;
+      return true;
   }
 
   NOTREACHED();
@@ -219,6 +222,12 @@ void PermissionServiceImpl::HasPermission(PermissionDescriptorPtr permission,
   std::move(callback).Run(GetPermissionStatus(permission));
 }
 
+bool PermissionServiceImpl::HasPermission(blink::mojom::PermissionDescriptorPtr permission,
+                                          blink::mojom::PermissionStatus* out_status) {
+  *out_status = GetPermissionStatus(permission);
+  return true;
+}
+
 void PermissionServiceImpl::RevokePermission(
     PermissionDescriptorPtr permission,
     PermissionStatusCallback callback) {
